src/modules/condition/then.rs

```diff
@@ -0,0 +1,120 @@
+            param_name: String::new(),
+            param_global_id: None
+        }
+    }
+
+    fn parse(&mut self, meta: &mut ParserMetadata) -> SyntaxResult {
+        match token(meta, "then") {
+            Ok(_) => {
+                let tok = meta.get_current_token();
+                
+                // Parse the parameter in parentheses
+                token(meta, "(")?;
+                let param_tok = meta.get_current_token();
+                
+                // Check if we immediately hit a closing paren (empty parameter)
+                if token(meta, ")").is_ok() {
+                    return error!(meta, param_tok, "Parameter name cannot be empty");
+                }
+                
+                let param_name = token_by(meta, |word| word.chars().all(|c| c.is_alphanumeric() || c == '_'))?;
+                
+                if param_name.is_empty() {
+                    return error!(meta, param_tok, "Parameter name cannot be empty");
+                }
+                
+                self.param_name = param_name;
+                token(meta, ")")?;
+                
+                // Add the parameter variable to the scope and parse the block
+                meta.with_push_scope(|meta| {
+                    self.param_global_id = meta.add_var(&self.param_name, Type::Num, false);
+                    syntax(meta, &mut *self.block)?;
+                    Ok(())
+                })?;
+                
+                if self.block.is_empty() {
+                    let message = Message::new_warn_at_token(meta, tok)
+                        .message("Empty then block")
+                        .comment("You should use 'trust' modifier to run commands without handling errors");
+                    meta.add_message(message);
+                }
+                self.is_parsed = true;
+                Ok(())
+            },
+            Err(_) => {
+                // If we're in a trust context, mark as parsed
+                if meta.context.is_trust_ctx {
+                    self.is_parsed = true;
+                }
+                // Otherwise, return quietly (no then block found)
+                Ok(())
+            }
+        }
+    }
+}
+
+impl TranslateModule for Then {
+    fn translate(&self, meta: &mut TranslateMetadata) -> FragmentKind {
+        if self.is_parsed {
+            // Create the parameter variable assignment using the proper variable name
+            let param_var_name = get_variable_name(&self.param_name, self.param_global_id);
+            let param_assignment = raw_fragment!("{}=$?", param_var_name);
+            
+            let block = self.block.translate(meta);
+            
+            match &block {
+                FragmentKind::Empty => {
+                    param_assignment.to_frag()
+                },
+                FragmentKind::Block(block) if block.statements.is_empty() => {
+                    param_assignment.to_frag()
+                },
```

Comment:
In these cases don't create the variable from parameter. The block is empty anyways, there is no logic that could utilize it.